File: seed/csharp-sdk/pagination/src/SeedPagination/Types/UsernameCursor.cs
using System.Text.Json.Serialization;
using SeedPagination.Core;

#nullable enable

namespace SeedPagination;

public record UsernameCursor
{
    [JsonPropertyName("cursor")]
    public required UsernamePage Cursor { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-model/pagination/src/UsernameCursor.php
<?php

namespace Seed;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;

class UsernameCursor extends SerializableType
{
    /**
     * @var UsernamePage $cursor
     */
    #[JsonProperty('cursor')]
    public UsernamePage $cursor;

    /**
     * @param array{
     *   cursor: UsernamePage,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->cursor = $values['cursor'];
    }
}
